modify declaration of dir variable in sections/dir

gurpreetatwal opened the pull request:
#### Description

potential fix for #420, feel free to close this PR if a better solution can be found

Fixes the issue by explicitly assigning the variable to prevent it from accidentally expanding the dir variable if there is a shell alias assigned to it.


Fix #420

denysdovhan commented on sections/dir.zsh:
```diff
@@ -21,7 +21,7 @@ SPACESHIP_DIR_COLOR="${SPACESHIP_DIR_COLOR="cyan"}"
   [[ $SPACESHIP_DIR_SHOW == false ]] && return
 
-  local dir
+  local 'dir'
```
Please leave a comment about using quotes here.

salmanulfarzy replied:
This should be default behavior for all unassigned local variables. Documenting it in `doc` would be better than leaving a single comment.

- http://mywiki.wooledge.org/Quotes
- http://zsh.sourceforge.net/Doc/Release/Shell-Grammar.html#Aliasing

gurpreetatwal commented:
@denysdovhan @salmanulfarzy rebased and cleaned up the PR
- modified all instances of `local <variable>` to be `local '<variable'>`
- updated example in API to link to the Zsh alias docs and include example of quoted local variable

gurpreetatwal commented on sections/venv.zsh:
```diff
@@ -27,7 +27,7 @@ spaceship_venv() {
   [ -n "$VIRTUAL_ENV" ] || return
 
-  local venv
+  local 'venv'
```
I could see this as being troublesome as well

example:
```zsh
alias -g venv=python3 -m venv
````
